hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java

```diff
@@ -122,4 +131,20 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         It will not be committed to state until commit is called on the state.--- */
         fileStore.put(fileBuilder.build());
     }
+
+    private void handleAppendUpgradeFile(FileAppendTransactionBody fileAppend, HandleContext handleContext) {
+        final var fileStore = handleContext.writableStore(WritableUpgradeStore.class);
+        File file = fileStore.peek();
+        if (file == null) {
+            throw new HandleException(INVALID_FILE_ID);
+        }
+        final var appendedFile = new File.Builder()
```

Comment:
I don't think it makes sense to store the `0.0.150` metadata in **each `File` part inside the `FCQueue`.**

Imo we should have a normal `KVState` entry for the `0.0.150` metadata; and then just keep its _data_ in the `WritableUpgradeStore`.

Reply:
It is relevant to the top comment I mentioned. It adds a lot of complexity, as I mentioned above. It will make it hard to understand that it is stored in two different places.